checks/pinned_dependencies.go

```diff
@@ -121,87 +121,123 @@ func maxScore(s1, s2 int) int {
 	return s2
 }
 
-func createReturnValues(r bool, infoMsg string, dl checker.DetailLogger, err error) (int, error) {
+type pinnedResult int
+
+const (
+	pinnedUndefined pinnedResult = iota
+	pinned
+	notPinned
+)
+
+func addPinnedResult(r *pinnedResult, to bool) {
+	// If the result is `notPinned`, we keep it.
+	// In other cases, we always update the result.
+	if *r == pinnedUndefined || *r == pinned {
```

Comment:
Nit: update to:

```
if r == nil || *r == notPinned {
  return
}
switch to {
...
}
```

Reply:
done. I don't catch the `r = nil` case because I'd rather have it panic and tell me something is wrong in the code than gracefully returning.